### PR TITLE
Add stats

### PR DESCRIPTION
**Pull request by shaunthium**

### CODE REVIEW

File: app/models/concerns/statistics/companies.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    def reviews_count
+      num_product_reviews = 0
+      self.products.each do |pdt|
+        num_product_reviews += pdt.reviews.count
+      end
+      num_service_reviews = 0
+      self.services.each do |svc|
+        num_service_reviews += svc.reviews.count
+      end
+      num_product_reviews + num_service_reviews
```
Comment (strengthandwill) [inline app/models/concerns/statistics/companies.rb]: Maybe can simplify with `reduce`? e.g.
```
products.reduce(0) { |s, p| s + p.reviews.count } + services.reduce(0) { |s, s| s + s.reviews.count }
```

File: app/models/concerns/statistics/products_and_services.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+  included do
+    def reviews_count
+      self.reviews.count
+    end
+
+    def aggregate_score
+      score = 0
+      if self.reviews.count > 0
+        self.reviews.each { |r| score += r.score }
```
Comment (strengthandwill) [inline app/models/concerns/statistics/products_and_services.rb]: Maybe can simplify using `sum`? e.g.
```
score = reviews.sum(:&score)
```

File: app/models/concerns/statistics/companies.rb (hunk 1 above)
Comment (shaunthium) [reply app/models/concerns/statistics/companies.rb]: Updated!

File: app/models/concerns/statistics/products_and_services.rb (hunk 2 above)
Comment (shaunthium) [reply app/models/concerns/statistics/products_and_services.rb]: Updated!

Comment (strengthandwill) [reply app/models/concerns/statistics/products_and_services.rb]: In the future for scaling and optimisation, can consider to save `aggregate_score`, and `reviews_count` as fields in the model and update the values only when `after_save` of the respective model, as computing these values for each requests maybe quite computational intensive when the volume is high, but for POC this is good enough;) This also allow sorting by fields when querying.

Comment (shaunthium) [reply app/models/concerns/statistics/products_and_services.rb]: Understood, thanks kah kong!

**Pull request merged**